lib/server.js

```diff
@@ -68,64 +67,111 @@ module.exports = Server;
  * @param {string} [opts.username] Username for RabbitMQ.
  * @param {string} [opts.password] Username for Password.
  */
-function Server(opts) {
+function Server (opts) {
+  this._tasks = {};
   this.opts = opts || {};
-  defaults(this.opts, Server.DEFAULT_OPTIONS);
+  defaults(this.opts, {
+    hostname: process.env.RABBITMQ_HOSTNAME || 'localhost',
+    port: process.env.RABBITMQ_PORT || 5672,
+    username: process.env.RABBITMQ_USERNAME || 'guest',
+    password: process.env.RABBITMQ_PASSWORD || 'guest'
+  });
 
-  if (!Array.isArray(this.opts.queues)) {
-    throw new Error('ponos.Server: missing required `queues` option.');
-  }
+  if (this.opts.hermes) {
+    this.hermes = this.opts.hermes;
+  } else {
+    if (!Array.isArray(this.opts.queues)) {
+      throw new Error('ponos.Server: missing required `queues` option.');
```

Comment:
This one is tricky because it's in the constructor. The _service_ that is created using `ponos` will most likely _also_ have logging, and if it's failing to even create the server, I imagine it will catch this 'missing options' error and report it itself (or a health check fails, etc). All that to say, I don't think this should _report_ the error yet, because the server can't even be initialized